src/Core/SettingsEnums.cpp

```diff
@@ -42,6 +43,37 @@ IMPLEMENT_SETTING_MULTI_ENUM(JoinAlgorithm, ErrorCodes::UNKNOWN_JOIN,
      {"grace_hash",           JoinAlgorithm::GRACE_HASH}})
 
 
+IMPLEMENT_SETTING_MULTI_ENUM(QueryKind, ErrorCodes::UNKNOWN_QUERY_KIND,
```

Comment:
I don't like that we now have two lists of query kinds... I mean this one and `IAST::QueryKind`...